SWDEV-556212 - Update changelog for HIP 7.1 in develop (#1326)
ROCm/rocm-systems: projects/clr/CHANGELOG.md

Full documentation for HIP is available at [rocm.docs.amd.com](https://rocm.docs.amd.com/projects/HIP/en/latest/index.html)

## HIP 7.2 for ROCm 7.2

### Added

* New HIP APIs
    - `hipLibraryLoadData`      Create library object from code
    - `hipLibraryLoadFromFile`  Create library object from file
    - `hipLibraryUnload`        Unload library
    - `hipLibraryGetKernel`     Get a kernel from library
    - `hipLibraryGetKernelCount` Get kernel count in library

## HIP 7.1 for ROCm 7.1

### Added

* New HIP APIs
    - `hipModuleGetFunctionCount` returns the number of functions within a module
    - `hipMemsetD2D8` used for setting 2D memory range with specified 8-bit values
    - `hipMemsetD2D8Async` used for setting 2D memory range with specified 8-bit values asynchronously
    - `hipMemsetD2D16` used for setting 2D memory range with specified 16-bit values
    - `hipMemsetD2D16Async` used for setting 2D memory range with specified 16-bit values asynchronously
    - `hipMemsetD2D32` used for setting 2D memory range with specified 32-bit values

>     - `hipMemsetD2D32Async` used for setting 2D memory range with specified 32-bit values asynchronously

Reviewer: ```suggestion
    - `hipMemsetD2D8` sets 2D memory range with specified 8-bit values
    - `hipMemsetD2D8Async` asynchronously sets 2D memory range with specified 8-bit values asynchronously
    - `hipMemsetD2D16` sets 2D memory range with specified 16-bit values
    - `hipMemsetD2D16Async` asynchronously sets 2D memory range with specified 16-bit values asynchronously
    - `hipMemsetD2D32` sets 2D memory range with specified 32-bit values
    - `hipMemsetD2D32Async` asynchronously sets 2D memory range with specified 32-bit values asynchronously
```

Author: Thanks. Updated.

    - `hipStreamSetAttribute` sets attributes such as synchronization policy for a given stream
    - `hipStreamGetAttribute` returns attributes such as priority for a given stream
    - `hipModuleLoadFatBinary`  loads fatbin binary to a module
    - `hipMemcpyBatchAsync` performs a batch of 1D or 2D memory copied asynchronously

>     - `hipMemcpy3DBatchAsync` performs a batch of 3D memory copied asynchronously

Reviewer: ```suggestion
    - `hipMemcpyBatchAsync` asynchronously performs a batch copy of 1D or 2D memory
    - `hipMemcpy3DBatchAsync` asynchronously performs a batch copy of 3D memory
```

Author: Thanks. Updated.

    - `hipMemcpy3DPeer` copies memory between devices

>     - `hipMemcpy3DPeerAsync`copied memory between devices asynchronously

Reviewer: ```suggestion
    - `hipMemcpy3DPeerAsync` asynchronously copies memory between devices 
```

Author: Thanks. Updated.

    - `hipMemsetD2D32Async` used for setting 2D memory range with specified 32-bit values

>       asynchronously

Reviewer: ```suggestion
    - `hipMemsetD2D32Async` asynchronously sets 2D memory range with specified 32-bit values
```

Author: Updated. Thanks.
